File: cosmowcl/src/app/shared/services/formato-file.service.ts
/*
 * Copyright CSI-Piemonte - 2023
 * SPDX-License-Identifier: GPL-3.0-or-later
 */
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { FormatoFile } from '../models/api/cosmoecm/formatoFile';
import { FormatoFileResponse } from '../models/api/cosmoecm/formatoFileResponse';
import { RaggruppamentoFormatiFileResponse } from '../models/api/cosmoecm/raggruppamentoFormatiFileResponse';
import { ApiUrls } from '../utilities/apiurls';

@Injectable()
export class FormatoFileService {

  constructor(private httpClient: HttpClient) { }

  /** GET: Restituisce una lista di custom form */
  getFormatiFile(filter: string): Observable<FormatoFileResponse> {
    const options = filter ?
      { params: new HttpParams().set('filter', filter) } : {};
    return this.httpClient.get<FormatoFileResponse>(ApiUrls.FORMATI_FILE, options);
  }

  getGroupedFormatiFile(filter: string): Observable<RaggruppamentoFormatiFileResponse> {
    const options = filter ?
      { params: new HttpParams().set('filter', filter) } : {};
    return this.httpClient.get<RaggruppamentoFormatiFileResponse>(ApiUrls.FORMATI_FILE_GROUPED, options);
  }
}

File: cosmoecm/src/java/it/csi/cosmo/cosmoecm/business/service/FormatoFileService.java
/*
 * Copyright CSI-Piemonte - 2023
 * SPDX-License-Identifier: GPL-3.0-or-later
 */

package it.csi.cosmo.cosmoecm.business.service;

import it.csi.cosmo.cosmoecm.dto.rest.FormatoFileResponse;
import it.csi.cosmo.cosmoecm.dto.rest.RaggruppamentoFormatiFileResponse;

/**
 *
 */

public interface FormatoFileService {

  /**
   * Metodo che restituisce i formati file
   *
   * @param filter filtro
   * @return una lista di formati file
   */
  FormatoFileResponse getFormatiFile(String filter);

  RaggruppamentoFormatiFileResponse getFormatiFileRaggruppati(String filter);

}
